src/irr/video/COpenGLPipelineLayout.h

```diff
@@ -0,0 +1,73 @@
+#ifndef __IRR_C_OPENGL_PIPELINE_LAYOUT_H_INCLUDED__
+#define __IRR_C_OPENGL_PIPELINE_LAYOUT_H_INCLUDED__
+
+#include "irr/video/IGPUPipelineLayout.h"
+#include "COpenGLExtensionHandler.h"
+
+namespace irr {
+namespace video
+{
+
+class COpenGLPipelineLayout : public IGPUPipelineLayout
+{
+public:
+    struct SMultibindParams
+    {
+        struct SFirstCount
+        {
+            GLuint first = 0u;
+            GLsizei count = 0u;
+        };
+
+        SFirstCount ubos;
+        SFirstCount ssbos;
+        SFirstCount textures;
+        SFirstCount textureImages;
+    };
+
+    COpenGLPipelineLayout(
+        const SPushConstantRange* const _pcRangesBegin, const SPushConstantRange* const _pcRangesEnd,
+        core::smart_refctd_ptr<IGPUDescriptorSetLayout> _layout0, core::smart_refctd_ptr<IGPUDescriptorSetLayout> _layout1,
+        core::smart_refctd_ptr<IGPUDescriptorSetLayout> _layout2, core::smart_refctd_ptr<IGPUDescriptorSetLayout> _layout3
+    ) : IGPUPipelineLayout(_pcRangesBegin, _pcRangesEnd, _layout0, _layout1, _layout2, _layout3)
+    {
+        SMultibindParams params;
+
+        for (size_t i = 0u; m_descSetLayouts.size(); ++i) {
+            IGPUDescriptorSetLayout* descSetLayout = m_descSetLayouts[i].get();
+
+            auto reset = [](SMultibindParams::SFirstCount& _fc) {
+                _fc.first = _fc.count;
+                _fc.count = 0u;
+            };
+
+            auto bindings = descSetLayout->getBindings();
+            for (const auto& bnd : bindings) {
+                if (bnd.type == asset::EDT_UNIFORM_BUFFER)
+                    params.ubos.count += bnd.count;
+                else if (bnd.type == asset::EDT_STORAGE_BUFFER)
+                    params.ssbos.count += bnd.count;
+                else if (bnd.type == asset::EDT_COMBINED_IMAGE_SAMPLER || bnd.type == asset::EDT_UNIFORM_TEXEL_BUFFER)
+                    params.textures.count += bnd.count;
```

Comment:
missing storage image and storage texel buffer

Reply:
resolved